src/Features/VisualBasic/Portable/Structure/Providers/MultiLineIfBlockStructureProvider.vb

```diff
@@ -14,9 +14,59 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Structure
                                                   spans As ArrayBuilder(Of BlockSpan),
                                                   options As OptionSet,
                                                   cancellationToken As CancellationToken)
-            spans.AddIfNotNull(CreateBlockSpanFromBlock(
-                               node, node.IfStatement, autoCollapse:=False,
-                               type:=BlockTypes.Conditional, isCollapsible:=True))
+
+            ' The full block: eg If ... Then ... End If
+            spans.AddIfNotNull(CreateBlockSpanFromBlock(node, node.IfStatement,
+                                                        autoCollapse:=False,
+                                                        type:=BlockTypes.Conditional,
+                                                        isCollapsible:=True))
+
+            ' Attempt to make the code block after the If...Then collapsable.
+            Make_IfThenCodeBlockCollapsible(node, spans)
+
+            ' Make the ElseIfs collapsable.
+            Make_ElseIfCodeBlocksCollapsible(node, spans)
+
+            ' Make the Else section collapsable.
+            Make_ElseCodeBlockCollapsible(node, spans)
+
+        End Sub
+
+        Private Shared Sub Make_ElseCodeBlockCollapsible(node As MultiLineIfBlockSyntax, spans As ArrayBuilder(Of BlockSpan))
+            If node.ElseBlock IsNot Nothing Then
+                spans.AddIfNotNull(CreateBlockSpanFromBlock(node.ElseBlock, "Else ... ",
+                                                            autoCollapse:=False,
+                                                            type:=BlockTypes.Conditional,
+                                                            isCollapsible:=True))
+            End If
+        End Sub
+
+        Private Shared Sub Make_ElseIfCodeBlocksCollapsible(node As MultiLineIfBlockSyntax, spans As ArrayBuilder(Of BlockSpan))
+            For Each [ElseIf] In node.ElseIfBlocks
+                spans.AddIfNotNull(CreateBlockSpanFromBlock([ElseIf], [ElseIf].ElseIfStatement,
+                                                            autoCollapse:=False,
+                                                            type:=BlockTypes.Conditional,
+                                                            isCollapsible:=True))
+            Next
+        End Sub
+
+        Private Shared Sub Make_IfThenCodeBlockCollapsible(node As MultiLineIfBlockSyntax, spans As ArrayBuilder(Of BlockSpan))
+            Dim statements = node.Statements
+            If statements.Count > 0 Then
+                Dim starting = node.Statements(0).GetLeadingTrivia.Where(Function(t) t.IsWhitespaceOrEndOfLine Or t.IsRegularOrDocComment)
+                Dim start = starting.Skip(1).FirstOrDefault.SpanStart
+                Dim finish = statements.Last.FullSpan.End - 2
```

Comment:
what is this -2 for?

Reply:
@CyrusNajmabadi  This is a bit of kludge, as it remove the newline trivia from the end of span. 
The actual span need, first line of the if then codeblock (ie the line after `Then` to either the end of the line before the first `ElseIf` or `Else`. 
`Span` doesn't include the comments
`FullSpan` does include the comments but also fetches the newlines at the start and finish, which results in the collapsed form being directly after the `Then` on the same line.

Better solutions welcome

Reply:
@CyrusNajmabadi 
It also doesn't help that the 
```VB
If ... Then
  '  ...
```
is a different class like `ElseIf` and `Else`.

Reply:
Yeah, the -2 won't work at all.  For all you know, there is no newline at the end. And the newline may not be 2-chars.  If you want up to the end without the newline, you'll have to actually trim things properly.